lib/active_model/serializer/configuration.rb

```diff
@@ -29,6 +29,28 @@ def config.array_serializer
         config.jsonapi_include_toplevel_object = false
         config.key_transform = nil
 
+        # example output
+        # => [
+        # "::Api::V1::ResNamespace::ResourceSerializer",
+        # "::Api::ResNamespace::ResourceSerializer",
+        # "::ResNamespace::ResourceSerializer" ,
+        # "::ResourceSerializer"]
+        config.serializer_lookup = Proc.new { |controller_name|
+          lookups = []
+          current_resource = controller_name.gsub(/Controller\z/, '')
+          current_resource += 'Serializer'
+          namespaces = current_resource.split('::')
+          resource_serializer = namespaces.slice!(-1)
+
+          lookups << '::' + resource_serializer
+          namespaces.length.times do |i|
+            current_namespace = namespaces[i - namespaces.length..-1].join('::')
+            lookups << '::' + current_namespace + '::' + resource_serializer
+          end
+
+          lookups
```

Comment:
Is that what we want to recommend to people?

I think it's pretty straightforward to do this, but maybe it could be documented somewhere? So people who ask can work with something? Idk 

The other thing I was thinking, is to Lee this option nil, so existing behavior remains the same, and provide the option? 

But yeah, I'm on the fence on if ams should even provide the capability to configure the lookup.

Maybe, if rails pushes more API centric usage, it would wake more sense?